pkg/services/ngalert/api/api_ruler.go

```diff
@@ -578,5 +578,8 @@ func (srv RulerSrv) searchAuthorizedAlertRules(ctx context.Context, c *contextmo
 }
 
 func getNamespaceKey(namespace *folder.Folder) string {
+	if namespace.ParentUID == "" {
+		return namespace.Title
+	}
```

Comment:
I think we should still return the leading slash and UI should strip it. This will simplify logic on the front end. Without the leading slash a folder title like "My Folder / Alerts" would could be treated by UI as a nested folder. 
However, if we return a folder like "/My Folder / Alerts" the UI can always treat the first slash as a folder separator. 

Alternative, is to escape slashes in the `namespace.Title`

Reply:
I think I prefer the alternative to escape slashes in the `namespace.Title` because the other (including the leading slash) would introduce bigger breaking changes when [provisioning to folders](https://github.com/grafana/grafana/pull/77450) (it would require users to modify all there existing provisioning configurations and add the leading slash)